backend/src/akvo/lumen/auth.clj

```diff
@@ -2,35 +2,61 @@
   (:require [akvo.commons.jwt :as jwt]
             [cheshire.core :as json]
             [clj-http.client :as client]
+            [clojure.set :as set]
             [clojure.string :as s]
             [ring.util.response :as response]))
 
+(defn claimed-roles [jwt-claims]
+  (set (get-in jwt-claims ["realm_access" "roles"])))
+
+(defn tenant-user?
+  [{:keys [tenant jwt-claims]}]
+  (contains? (claimed-roles jwt-claims)
+             (format "akvo:lumen:%s" tenant)))
+
+(defn tenant-admin?
+  [{:keys [tenant jwt-claims]}]
+  (contains? (claimed-roles jwt-claims)
+             (format "akvo:lumen:%s:admin" tenant)))
+
+(defn public-path? [{:keys [path-info request-method]}]
+  (and (= :get request-method)
+       (or (= "/api" path-info)
+           (= "/env" path-info)
+           (s/starts-with? path-info "/s/")
```

Comment:
I think something happened with the merge from `develop` here as this should be `/share/`